src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Util.java

```diff
@@ -87,14 +87,19 @@ private static Provider getProvider(Provider p, String providerName,
         }
         p = Security.getProvider(providerName);
         if (p == null) {
-            try {
-                @SuppressWarnings("deprecation")
-                Object o = Class.forName(className).newInstance();
-                p = (Provider)o;
-            } catch (Exception e) {
-                throw new ProviderException
-                        ("Could not find provider " + providerName, e);
-            }
+            p = AccessController.doPrivileged(
+                new PrivilegedAction<Provider>() {
+                    public Provider run() {
+                        try {
+                            @SuppressWarnings("deprecation")
+                            Object o = Class.forName(className).newInstance();
+                            return (Provider) o;
+                        } catch (Exception e) {
+                            throw new ProviderException
+                                    ("Could not find provider " + providerName, e);
+                        }
+                    }
+                });
```

Comment:
Sean's suggestion is to add additional arguments here, e.g. null, new RuntimePermission("accessClassInPackage." + <pkgName>).

Reply:
Yes, replied to Sean above. Thanks.